src/core/dynamic_batch_scheduler.cc

```diff
@@ -349,32 +353,63 @@ DynamicBatchScheduler::SchedulerThread(
                  << "...";
 }
 
-void
-DynamicBatchScheduler::InitPendingShape(const InferRequestHeader& request)
+Status
+DynamicBatchScheduler::InitPendingShape(
+    const std::shared_ptr<InferRequestProvider>& request_provider)
 {
   pending_batch_shapes_.clear();
 
+  const InferRequestHeader& request = request_provider->RequestHeader();
   for (const auto& input : request.input()) {
-    pending_batch_shapes_.emplace(std::make_pair(input.name(), input.dims()));
+    const auto itr = enforce_equal_shape_tensors_.find(input.name());
+    if (itr != enforce_equal_shape_tensors_.end()) {
+      std::pair<DimsList, std::vector<int64_t>> shapes;
+      shapes.first = input.dims();
+
+      // For shape tensors must compare the contents of the tensor in
+      // addition to the tensor shape itself.
+      if (itr->second) {
+        //        RETURN_IF_ERROR(
+        //            request_provider->PeekShapeTensor(input, &shapes.second));
+      }
+
+      pending_batch_shapes_.emplace(
+          std::make_pair(input.name(), std::move(shapes)));
```

Comment:
So as per out discussion, for protecting all these PeekShapeTensor() calls shape tensors must set their respective enforce_equal_shape_tensors_ to true. And this information has to come from the model configuration. Specifically, the IsShapeTensor() must access the model configuration to return the correct property?

Reply:
Yes, once the is_shape_tensor is available from the model configuration can re-implement/remove the existing IsShapeTensor() call. enforce_equal_stape_tensors_ isn't a bool, but yes it will be set based upon IsShapeTensor() and if there are not shape tensors the PeekShapeTensor will not be called.